fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/InternalPlannerGraphRewritable.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * InternalPlannerGraphRewritable.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.temp;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Interface to allow {@link PlannerExpression}s to rewrite their own
+ * internal graph representation.
+ */
+public interface InternalPlannerGraphRewritable {
+    @Nonnull
+    InternalPlannerGraph rewriteInternalPlannerGraph(@Nonnull List<InternalPlannerGraph> childGraphs);
```

Comment:
I think that this needs a detailed Javadoc explaining what it does (and especially what `childGraphs` is). In particular, this is important because the interface shows up in so many `RecordQueryPlan` implementations.